### ISSUE TITLE
Header width 'wrap' content

### ISSUE THREAD

**Issue by rajsuvariya** [react-native-community/react-native-tab-view#323]
I want to have the tabs and a viewpager below it. This is a nice library and I was able to implement the same. Now according to my app requirement. The tab should not occupy the entire width of the window. It should have width equal to wrap the content. I was able to do it by providing `tabStyle = {{ flex: 0 }}`. But the issue is tha indicator is still showing for the tabsize with `flex=1`. How can I give the same width to the indicator as tab???

**Comment by ixje** [comment]
I'm not sure if I fully understand your text, but if you want to change the indicator then have a look at the `indicatorStyle` prop.

**Comment by rajsuvariya** [comment]
![image](https://user-images.githubusercontent.com/12663795/29061154-40c58f0c-7c3a-11e7-9828-00b172f9c4ff.png)

I want smething like this

I tried this code


   ```
 _renderHeader = props => <TabBar {...props}
      style={{ height: 35, backgroundColor:'white'}}
      indicatorStyle={{ height: 0, borderBottomWidth:2, borderBottomColor:'#f7a500' }}
      getLabelText={({route})=>route.title}
      tabStyle={{ flex: 0, alignItems: 'center',  justifyContent: 'center', padding: 0 }}
      labelStyle={{ color: 'black', fontSize:16 }}/>;

      _renderScene = SceneMap({
        '1': SecondRoute,
        '2': FirstRoute,
      });

    render() {
      return(
        <Card>
          <TabViewAnimated
            style={{ flex: 1 }}
            navigationState={this.state}
            renderScene={this._renderScene}
            renderHeader={this._renderHeader}
            onIndexChange={this._handleIndexChange}
          />
        </Card>
      );
    }
```
I am able to get the tab lable at right position but the indicator is not wrapping with respect to tab size.

![image](https://user-images.githubusercontent.com/12663795/29061214-8c8eadb0-7c3a-11e7-827f-9dc46851016a.png)

**Comment by rajsuvariya** [comment]
@ixje any solution for this?

**Comment by ixje** [comment]
@rajsuvariya Aha I understand your need. I looked into it a bit but I don't think it's currently doable via any props without modifications. The indicator width is determined by the tab width and then can be overruled by a `width` in the `indicatorStyle` prop (see below). 

**TabBar.js**
```
const finalTabWidth = this._getFinalTabWidth(this.props)

                    {this._renderIndicator({
                        ...this.props,
                        width: new Animated.Value(finalTabWidth)
                    })}
```

```
    _getFinalTabWidth = (props: Props<T>) => {
        const { layout, navigationState } = props
        const tabWidth = this._getTabWidthFromStyle(props.tabStyle)
        if (typeof tabWidth === 'number') {
            return tabWidth
        }
        if (typeof tabWidth === 'string' && tabWidth.endsWith('%')) {
            return layout.width * (parseFloat(tabWidth) / 100)
        }
        if (props.scrollEnabled) {
            return layout.width / 5 * 2
        }
        return layout.width / navigationState.routes.length
    }
```

However, even if we conditionally set the indicator width the indicator position won't update based on these props as far as I could see (and tested).

**Comment by rajsuvariya** [comment]
Yes I also tried to modify the library but was not able to achieve it

**Comment by orendecor** [comment]
We faced the same problem in our app and we solved statically by calculate the container width and devide it by the number of tabs, something like this:


```
const CONTAINER_WIDTH = 300; //or other calculation like: Dimensions.get('window').width - x
const NUMBER_OF_TABS = 3;

<TabBar style={{flex:0,flexBasis: CONTAINER_WIDTH}}
                tabStyle={CONTAINER_WIDTH /NUMBER_OF_TABS } 
```

**Comment by smithaitufe** [comment]
I have the same need. Was anyone able to find a solution for this?

**Comment by toytonic** [comment]
Same issue here :)

My tabbar should have a different width than the actual tab pager container. Currently it's very hard to achieve without a custom tabbar, which in my case will reimplement most of the original behaviour.

@satya164 is there any specific reason why the indicator width and position are calculated based on the tab pager container and not the tabbar (container) width?

**Comment by pribeh** [comment]
Also in need of a decent solution to this.

**Comment by BhaumikSurani** [comment]
I'm some change in TabBar.js and it is working

![small](https://user-images.githubusercontent.com/34732349/43306109-a4294fac-9197-11e8-900c-3e2625304f5e.png)
![midum](https://user-images.githubusercontent.com/34732349/43306072-88741fe4-9197-11e8-9e58-07e0b3698571.png)
![large](https://user-images.githubusercontent.com/34732349/43306083-90e1cdfc-9197-11e8-9773-c9a5a9998d03.png)

Add Following Line
`	isAutoSizeIndicator?: boolean, `
inside 
`	type Props<T> = SceneRendererProps<T> & {......}`

Add Following Line
`	isAutoSizeIndicator: PropTypes.bool,`
inside
`	static propTypes = {......}`	

Add following code inside **constructor(props: Props<T>) {.....}**
```
this.indicatorWidth = new Array();
for (var i = 0; i<this.props.navigationState.routes.length; i++) {
    this.indicatorWidth.push(0);
};
```
Add Following Method
```
_renderAutoWidthIndicator = (props: IndicatorProps<T>) => {
	if (typeof this.props.renderIndicator !== 'undefined') {
	  return this.props.renderIndicator(props);
	}
	const { width, position, navigationState } = props;
	var translateX=0;
	for (var i=0; i<navigationState.index ; i++) {
	  translateX= translateX + this.indicatorWidth[i];
	};
	return (
	  <Animated.View
	    style={[
	      styles.indicator,
	      { width, transform: [{ translateX }] },
	      this.props.indicatorStyle,
	    ]} />
	);
};
```
**Then Some Change in render() method**

**First Change**
```
<Animated.View
  pointerEvents="none"
  style={[
    styles.indicatorContainer,
    scrollEnabled
      //? { width: tabBarWidth, transform: [{ translateX }] }
      ? { width:this.indicatorWidth[navigationState.index] , transform: [{ translateX }] }
      : null,
  ]}
>
  {!this.props.isAutoSizeIndicator && this._renderIndicator({
    ...this.props,
    width: tabWidth,
  })}

  {this.props.isAutoSizeIndicator && this._renderAutoWidthIndicator({
    ...this.props,
    width: this.indicatorWidth[navigationState.index],
  })}
</Animated.View>
```

**Second Change**
```
<View pointerEvents="none" style={styles.container}
onLayout={(event) => {
  var {x, y, width, height} = event.nativeEvent.layout;
  this.indicatorWidth[i]=width;
}}>
	........
</View>
```


When You want to use pass props like Following inside TabView
```
<TabView
	.........
	<TabBar
		{...props}
		tabStyle={{width:'auto'}}
		isAutoSizeIndicator={true} />
/>
```

**Comment by rochapablo** [comment]
any update about this?

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.

**Comment by bookjan** [comment]
Hi, I share an example of the indicator width wrap the content width.

![Aug-08-2019 11-42-07](https://user-images.githubusercontent.com/1734406/62674985-70051800-b9d7-11e9-92c7-141b46423d68.gif)

The example code you can get it from [here](https://gist.github.com/bookjan/412d88ebd6b98f8ed49900be946668bf)
